src/libraries/Microsoft.Extensions.Configuration.Xml/src/XmlDocumentDecryptor.cs

```diff
@@ -84,17 +85,21 @@ public XmlReader CreateDecryptingXmlReader(Stream input, XmlReaderSettings setti
         /// <returns>An XmlReader which can read the document.</returns>
         protected virtual XmlReader DecryptDocumentAndCreateXmlReader(XmlDocument document)
         {
+#if NET
+            if (OperatingSystem.IsBrowser())
+            {
+                throw new PlatformNotSupportedException();
```

Comment:
No need to add `[UnsupportedOSPlatform("browser")]`?

Reply:
I wasn't sure how to do this. This method gets called when you have an encrypted XML file, and it doesn't get called when you don't. So depending on the data passed in, it will throw or not on browser.

https://github.com/dotnet/runtime/blob/12a8819eee9865eb38bca6c05fdece1053102854/src/libraries/Microsoft.Extensions.Configuration.Xml/src/XmlDocumentDecryptor.cs#L68-L71

Reply:
I see, thanks for the explanation, the dependent XML document is populated within the API, not passed to the API, therefore I don't think it could be covered by [Conditionally-Supported Platform Annotations](https://github.com/dotnet/runtime/issues/44922) proposal we might add to the analyzer in the future

Reply:
I've updated this a bit.

1. I've marked the `protected virtual DecryptDocumentAndCreateXmlReader` method as unsupported on browser.
2. I've suppressed the upstream warning that calls it, since it depends on the input data. If this method gets invoked, it will just call into `EncryptedXml` and let it throw on browser.

PTAL

Reply:
That is great, now if anyone use `DecryptDocumentAndCreateXmlReader` they know it is not supported on the browser, thanks!